src/plugins/workspace/public/components/workspace_menu/workspace_menu.tsx

```diff
@@ -67,111 +81,92 @@ export const WorkspaceMenu = ({ coreStart }: Props) => {
     setPopover(false);
   };
 
-  const workspaceToItem = (workspace: WorkspaceObject) => {
-    const workspaceURL = formatUrlWithWorkspaceId(
-      coreStart.application.getUrlForApp(WORKSPACE_DETAIL_APP_ID, {
+  const workspaceToItem = (workspace: WorkspaceObject, itemType: string) => {
+    const appId = navGroupsMap![getUseCase(workspace)]?.navLinks[0].id;
```

Comment:
Prefer `const appId = navGroupsMap?.[getUseCase(workspace)]?.navLinks[0]?.id; `

Reply:
When `getUseCase(workspace)` return a undefined result, clicking the workspace name will navigate to workspace detail page. I will update the code.